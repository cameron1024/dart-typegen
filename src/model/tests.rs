use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
