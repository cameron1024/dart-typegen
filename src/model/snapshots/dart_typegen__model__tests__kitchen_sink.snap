---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    defaults: Some(
        Defaults {
            generate_to_string: Some(
                SpannedScalar {
                    value: true,
                    span: SourceSpan {
                        offset: SourceOffset(
                            134,
                        ),
                        length: 4,
                    },
                },
            ),
            union: Some(
                UnionOptions {
                    sealed: Some(
                        SpannedScalar {
                            value: true,
                            span: SourceSpan {
                                offset: SourceOffset(
                                    161,
                                ),
                                length: 4,
                            },
                        },
                    ),
                    json_discriminant: Some(
                        SpannedScalar {
                            value: "custom-discriminant",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    188,
                                ),
                                length: 21,
                            },
                        },
                    ),
                },
            ),
            field: Some(
                FieldOptions {
                    json_key_case: Some(
                        SpannedScalar {
                            value: Snake,
                            span: SourceSpan {
                                offset: SourceOffset(
                                    242,
                                ),
                                length: 7,
                            },
                        },
                    ),
                },
            ),
        },
    ),
    enums: [
        Enum {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Color",
                span: SourceSpan {
                    offset: SourceOffset(
                        662,
                    ),
                    length: 7,
                },
            },
            docs: None,
            variants: [
                EnumVariant {
                    name: SpannedScalar {
                        value: "red",
                        span: SourceSpan {
                            offset: SourceOffset(
                                682,
                            ),
                            length: 5,
                        },
                    },
                    docs: Some(
                        SpannedScalar {
                            value: "a bad color",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    699,
                                ),
                                length: 13,
                            },
                        },
                    ),
                    json_value: Some(
                        Value {
                            type_name: None,
                            literal: Spanned {
                                span: Span(
                                    729,
                                    740,
                                ),
                                value: String(
                                    "ew, gross",
                                ),
                            },
                        },
                    ),
                },
                EnumVariant {
                    name: SpannedScalar {
                        value: "green",
                        span: SourceSpan {
                            offset: SourceOffset(
                                755,
                            ),
                            length: 7,
                        },
                    },
                    docs: Some(
                        SpannedScalar {
                            value: "\nThe best color\n\nTruly, there has never been a greater color\n    ",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    774,
                                ),
                                length: 70,
                            },
                        },
                    ),
                    json_value: Some(
                        Value {
                            type_name: None,
                            literal: Spanned {
                                span: Span(
                                    861,
                                    864,
                                ),
                                value: Int(
                                    Integer(
                                        Dec,
                                        "123",
                                    ),
                                ),
                            },
                        },
                    ),
                },
                EnumVariant {
                    name: SpannedScalar {
                        value: "blue",
                        span: SourceSpan {
                            offset: SourceOffset(
                                880,
                            ),
                            length: 6,
                        },
                    },
                    docs: None,
                    json_value: None,
                },
            ],
            extra_dart: [
                SpannedScalar {
                    value: "void hello() {}",
                    span: SourceSpan {
                        offset: SourceOffset(
                            901,
                        ),
                        length: 17,
                    },
                },
                SpannedScalar {
                    value: "\n    void goodbye() => print(\"cya\");\n  ",
                    span: SourceSpan {
                        offset: SourceOffset(
                            932,
                        ),
                        length: 44,
                    },
                },
            ],
        },
    ],
    classes: [
        Class {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "TopLevel",
                span: SourceSpan {
                    offset: SourceOffset(
                        263,
                    ),
                    length: 10,
                },
            },
            fields: [
                Field {
                    name: SpannedScalar {
                        value: "name",
                        span: SourceSpan {
                            offset: SourceOffset(
                                284,
                            ),
                            length: 6,
                        },
                    },
                    ty: SpannedScalar {
                        value: "String",
                        span: SourceSpan {
                            offset: SourceOffset(
                                296,
                            ),
                            length: 8,
                        },
                    },
                    defaults_to: None,
                    defaults_to_dart: None,
                    docs: None,
                },
                Field {
                    name: SpannedScalar {
                        value: "age",
                        span: SourceSpan {
                            offset: SourceOffset(
                                313,
                            ),
                            length: 5,
                        },
                    },
                    ty: SpannedScalar {
                        value: "int",
                        span: SourceSpan {
                            offset: SourceOffset(
                                324,
                            ),
                            length: 5,
                        },
                    },
                    defaults_to: Some(
                        Value {
                            type_name: None,
                            literal: Spanned {
                                span: Span(
                                    390,
                                    393,
                                ),
                                value: Int(
                                    Integer(
                                        Dec,
                                        "123",
                                    ),
                                ),
                            },
                        },
                    ),
                    defaults_to_dart: None,
                    docs: Some(
                        SpannedScalar {
                            value: "the age of this toplevel thing",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    341,
                                ),
                                length: 32,
                            },
                        },
                    ),
                },
                Field {
                    name: SpannedScalar {
                        value: "color",
                        span: SourceSpan {
                            offset: SourceOffset(
                                407,
                            ),
                            length: 7,
                        },
                    },
                    ty: SpannedScalar {
                        value: "Color",
                        span: SourceSpan {
                            offset: SourceOffset(
                                420,
                            ),
                            length: 7,
                        },
                    },
                    defaults_to: None,
                    defaults_to_dart: Some(
                        SpannedScalar {
                            value: "Color.green",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    451,
                                ),
                                length: 13,
                            },
                        },
                    ),
                    docs: Some(
                        SpannedScalar {
                            value: "\nThe color of the thing\n\nDefaults to [Color.green] because it's the best color\n    ",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    474,
                                ),
                                length: 88,
                            },
                        },
                    ),
                },
                Field {
                    name: SpannedScalar {
                        value: "pet",
                        span: SourceSpan {
                            offset: SourceOffset(
                                576,
                            ),
                            length: 5,
                        },
                    },
                    ty: SpannedScalar {
                        value: "Animal",
                        span: SourceSpan {
                            offset: SourceOffset(
                                587,
                            ),
                            length: 8,
                        },
                    },
                    defaults_to: None,
                    defaults_to_dart: None,
                    docs: None,
                },
                Field {
                    name: SpannedScalar {
                        value: "secondPet",
                        span: SourceSpan {
                            offset: SourceOffset(
                                605,
                            ),
                            length: 11,
                        },
                    },
                    ty: SpannedScalar {
                        value: "Animal?",
                        span: SourceSpan {
                            offset: SourceOffset(
                                622,
                            ),
                            length: 9,
                        },
                    },
                    defaults_to: Some(
                        Value {
                            type_name: None,
                            literal: Spanned {
                                span: Span(
                                    646,
                                    650,
                                ),
                                value: Null,
                            },
                        },
                    ),
                    defaults_to_dart: None,
                    docs: None,
                },
            ],
            docs: None,
            extra_dart: [],
        },
    ],
    unions: [
        Union {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Unused",
                span: SourceSpan {
                    offset: SourceOffset(
                        1038,
                    ),
                    length: 8,
                },
            },
            sealed: Some(
                SpannedScalar {
                    value: false,
                    span: SourceSpan {
                        offset: SourceOffset(
                            1054,
                        ),
                        length: 5,
                    },
                },
            ),
            json_discriminant: Some(
                SpannedScalar {
                    value: "something-custom-for-unused",
                    span: SourceSpan {
                        offset: SourceOffset(
                            1082,
                        ),
                        length: 29,
                    },
                },
            ),
            docs: None,
            classes: [
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "X",
                        span: SourceSpan {
                            offset: SourceOffset(
                                1124,
                            ),
                            length: 3,
                        },
                    },
                    fields: [],
                    docs: None,
                    extra_dart: [],
                },
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Y",
                        span: SourceSpan {
                            offset: SourceOffset(
                                1136,
                            ),
                            length: 3,
                        },
                    },
                    fields: [],
                    docs: None,
                    extra_dart: [],
                },
            ],
            extra_dart: [],
        },
        Union {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Animal",
                span: SourceSpan {
                    offset: SourceOffset(
                        1148,
                    ),
                    length: 8,
                },
            },
            sealed: None,
            json_discriminant: None,
            docs: Some(
                SpannedScalar {
                    value: "\nSome docs for animal\n\nIt's either a dog or a cat\n  ",
                    span: SourceSpan {
                        offset: SourceOffset(
                            1166,
                        ),
                        length: 57,
                    },
                },
            ),
            classes: [
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Dog",
                        span: SourceSpan {
                            offset: SourceOffset(
                                1232,
                            ),
                            length: 5,
                        },
                    },
                    fields: [
                        Field {
                            name: SpannedScalar {
                                value: "name",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        1278,
                                    ),
                                    length: 6,
                                },
                            },
                            ty: SpannedScalar {
                                value: "String",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        1290,
                                    ),
                                    length: 8,
                                },
                            },
                            defaults_to: Some(
                                Value {
                                    type_name: None,
                                    literal: Spanned {
                                        span: Span(
                                            1319,
                                            1324,
                                        ),
                                        value: String(
                                            "Dog",
                                        ),
                                    },
                                },
                            ),
                            defaults_to_dart: None,
                            docs: Some(
                                SpannedScalar {
                                    value: "the name",
                                    span: SourceSpan {
                                        offset: SourceOffset(
                                            1336,
                                        ),
                                        length: 10,
                                    },
                                },
                            ),
                        },
                        Field {
                            name: SpannedScalar {
                                value: "color",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        1363,
                                    ),
                                    length: 7,
                                },
                            },
                            ty: SpannedScalar {
                                value: "Color",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        1376,
                                    ),
                                    length: 7,
                                },
                            },
                            defaults_to: None,
                            defaults_to_dart: None,
                            docs: None,
                        },
                    ],
                    docs: Some(
                        SpannedScalar {
                            value: "Fun-loving buddy",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    1249,
                                ),
                                length: 18,
                            },
                        },
                    ),
                    extra_dart: [],
                },
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Cat",
                        span: SourceSpan {
                            offset: SourceOffset(
                                1397,
                            ),
                            length: 5,
                        },
                    },
                    fields: [
                        Field {
                            name: SpannedScalar {
                                value: "name",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        1441,
                                    ),
                                    length: 6,
                                },
                            },
                            ty: SpannedScalar {
                                value: "String",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        1453,
                                    ),
                                    length: 8,
                                },
                            },
                            defaults_to: Some(
                                Value {
                                    type_name: None,
                                    literal: Spanned {
                                        span: Span(
                                            1482,
                                            1503,
                                        ),
                                        value: String(
                                            "Destroyer of Worlds",
                                        ),
                                    },
                                },
                            ),
                            defaults_to_dart: None,
                            docs: None,
                        },
                        Field {
                            name: SpannedScalar {
                                value: "satanicPower",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        1520,
                                    ),
                                    length: 14,
                                },
                            },
                            ty: SpannedScalar {
                                value: "int",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        1540,
                                    ),
                                    length: 5,
                                },
                            },
                            defaults_to: None,
                            defaults_to_dart: Some(
                                SpannedScalar {
                                    value: "9001",
                                    span: SourceSpan {
                                        offset: SourceOffset(
                                            1571,
                                        ),
                                        length: 6,
                                    },
                                },
                            ),
                            docs: None,
                        },
                    ],
                    docs: Some(
                        SpannedScalar {
                            value: "Evil hellspawn",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    1414,
                                ),
                                length: 16,
                            },
                        },
                    ),
                    extra_dart: [],
                },
            ],
            extra_dart: [],
        },
    ],
}
